src/lib/logstore/log_dev.cpp

```diff
@@ -580,7 +583,242 @@ void LogDev::rollback(logstore_id_t store_id, logid_range_t id_range) {
     m_logdev_meta.add_rollback_record(store_id, id_range, true);
 }
 
-void LogDev::get_status(const int verbosity, nlohmann::json& js) const {
+/////////////////////////////// LogStore Section ///////////////////////////////////////
+
+void LogDev::handle_unopened_log_stores(bool format) {
+    for (auto it{std::begin(m_unopened_store_io)}; it != std::end(m_unopened_store_io); ++it) {
+        LOGINFO("skip log entries for store id {}-{}, ios {}", m_logdev_id, it->first, it->second);
+    }
+    m_unopened_store_io.clear();
+
+    // If there are any unopened storeids found, loop and check again if they are indeed open later. Unopened log store
+    // could be possible if the ids are deleted, but it is delayed to remove from store id reserver. In that case,
+    // do the remove from store id reserver now.
+    // TODO: At present we are assuming all unopened store ids could be removed. In future have a callback to this
+    // start routine, which takes the list of unopened store ids and can return a new set, which can be removed.
+    {
+        folly::SharedMutexWritePriority::WriteHolder holder(m_store_map_mtx);
+        for (auto it{std::begin(m_unopened_store_id)}; it != std::end(m_unopened_store_id);) {
+            if (m_id_logstore_map.find(*it) == m_id_logstore_map.end()) {
+                // Not opened even on second time check, simply unreserve id
+                unreserve_store_id(*it);
+            }
+            it = m_unopened_store_id.erase(it);
+        }
+
+        // Also call the logstore to inform that start/replay is completed.
+        if (!format) {
+            for (auto& p : m_id_logstore_map) {
+                auto& lstore{p.second.m_log_store};
+                if (lstore && lstore->get_log_replay_done_cb()) {
+                    lstore->get_log_replay_done_cb()(lstore, lstore->seq_num() - 1);
+                    lstore->truncate(lstore->truncated_upto());
+                }
+            }
```

Comment:
I don't think this code needs to be here. This method is about handling unopened_ids and we are doing things which are opened ids as well here right? Can we move this block to the caller. Yes it needs to take the write lock twice, which is ok